Sources/MeiliSearch/Model/SearchParameters.swift

```diff
@@ -31,7 +31,14 @@ public struct SearchParameters: Codable, Equatable {
     public let attributesToHighlight: [String]
 
     /// Attribute with an exact match.
-    public let filters: Filter?
+    public let filters: String?
+
+    /// Select which attribute has to be filtered, useful when you need to narrow down the results of the filter.
+    //TODO: Migrate to FacetFilter object
+    public let facetFilters: [[String]]?
```

Comment:
Since i'm not familiar with swift types, it is possible your type `[[String]]` is already good. Just in case, since you asked for it, I will go deeper into `facetFilters`.

FacetFilters can go up to `2` of array deepness. 
It can either take:
- a string
- an array of string(s)
- an array of string(s) and other array(s) of string(s)

The following are correct: 

good ✅
```javascript
index.search("q", {
facetFilters: "genre:comedy"
}
```

good ✅
```javascript
index.search("q", {
facetFilters: ["genre"]
}
```

 good ✅
```javascript
index.search("q", {
facetFilters: ["genre:comedy", ["genre:horror"]]
}
```


When you add one more array deepness, it will raise errors: 
error ❌
```javascript
index.search("q", {
facetFilters: ["genre:comedy", ["genre:horror", ["genre:romance"]]]
}
```

error ❌
```javascript
index.search("q", {
facetFilters: [[["genre:romance"]]]
}
```

In typescript this is how I have declared them:

```javascript
export type FacetFilter = (string | string[])[]

export interface SearchParams {
  facetFilters?: string | FacetFilter | FacetFilter[]
}
```


_(i already showed you what I did in typescript but I had to change it as apparently doing recursivity in types is bad practice)_

Reply:
Thank you for the answer :)